pyhms/tree.py

```diff
@@ -227,3 +232,34 @@ def tree(self) -> str:
             + "\n"
             + format_deme_children_tree(self.root, best_fitness=self.best_individual.fitness)
         )
+
+    def save_animation(
+        self,
+        filepath: str = "hms_tree.gif",
+        dimensionality_reducer: DimensionalityReducer = NaiveDimensionalityReducer(),
+    ) -> None:
+        """
+        Saves an animation of the tree evolution as an gif/mp4 file (gif is used by default).
+        In case of multidimensional genomes dimensionaty reducer is employed.
+        By default it uses NaiveDimensionalityReducer which takes only the first two dimensions.
+        """
+        save_tree_animation(self, filepath, dimensionality_reducer)
+
+    def plot_problem(self) -> None:
+        """
+        Plots 2D grid for the root level problem.
+        In case of multidimensional problem, only the first two dimensions are considered.
+        """
+        grid = Grid2DProblemEvaluation(self.config.levels[0].problem, self.config.levels[0].bounds)
+        grid.evaluate()
+        grid.plot()
+
+    def plot_best_fitness(self) -> None:
```

Comment:
Both 'plot_best_fitness' and 'plot_problem_surface' methods only show the plot while 'save_animation' saves a result. In my opinion new methods should also allow saving the results. Overall the effect of this all visualization methods should be of similar type.

Reply:
The reason is the fact that it's not that easy to display gif in jupyter notebook. I improved these methods to enable saving results.